api/v1alpha1/workstatus_types.go

```diff
@@ -31,13 +31,21 @@ type WorkStatus struct {
 
 	Spec   WorkStatusSpec `json:"spec,omitempty"`
 	Status RawStatus      `json:"status,omitempty"`
+	StatusDetails StatusDetails `json:"statusDetails, omitempty"`
 }
 
 // Workstatus spec
 type WorkStatusSpec struct {
 	SourceRef SourceRef `json:"sourceRef,omitempty"`
 }
 
+// StatusDetails spec
+type StatusDetails struct {
+	LastGeneration int64                 `json:"lastGeneration,omitempty"`
+	LastGenerationIsApplied  bool        `json:"lastGenerationIsApplied,omitempty"`
+	LastCurrencyUpdateTime  *metav1.Time `json:"lastCurrencyUpdateTime,omitempty"`
```

Comment:
Do you really want these to be optional? Compare the declarations in https://github.com/kubestellar/kubestellar/pull/2457